### ISSUE TITLE
Auto-detect marker API support 

### ISSUE THREAD

**Issue by jrmadsen** [RRZE-HPC/likwid#307]
Hey @TomTheBear I have been working on [improving my support for likwid in timemory](https://github.com/NERSC/timemory/pull/74) lately bc I had previously required the user to add `-DLIKWID_PERFMON` and `-DLIKWID_NVMON` and I realized that this deviates from how I have set other things up. I added new cmake configuration options `TIMEMORY_USE_LIKWID_PERFMON` and `TIMEMORY_USE_LIKWID_NVMON` to support these being automatically added but I expect this is going to cause some additional complexities in the future in spack, etc., anybody building timemory from scratch at NERSC or another system where LIKWID is already installed (user would have to know how LIKWID was installed), and, honestly, the [number of CMake options is getting overwhelming](https://timemory.readthedocs.io/en/develop/installation.html) so I want to try to minimize things like this. 

So, basically, what I am looking for: a header generated by the configuration system that maybe looks like this:

```
#if defined(LIKWID_AUTO_DEFINE_MARKER_API)
#  define LIKWID_PERFMON
// #  define LIKWID_NVMON
#endif
```

So that one can just use a generic define (e.g. `-DLIKWID_AUTO_DEFINE_MARKER_API`) and auto-define the specific backends that the LIKWID installation has support for. CMake has a way to do this but unfortunately I am not familiar with your build system to create a PR (Also, it may already be available and I missed it somewhere).

**Comment by TomTheBear** [comment]
Hi,

This looks like a lot of progress in Timemory. Unfortunately, there is currently no such header and no other method in the C library that gives you this information. There is a `likwid.nvSupported()` in the Lua frontend, though. `LIKWID_PERFMON` is always supported, there is no switch to deactivate the CPU MarkerAPI. `LIKWID_NVMON` is only available when built with `NVIDIA_INTERFACE=true`.

For cmake, there is already the file `make/likwid-config.cmake` which is adjusted by the build system (installed to lib directory). It currently contains only the library and include paths but I could also add the flags. Is that suitable for Timemory?

Best,
Thomas

**Comment by jrmadsen** [comment]
Oh wow, I completely missed that you had that file. That is perfect. I can put in a PR for it so that `find_package(LIKWID)` just works and I'll add the support for what I need so that `find_package(LIKWID REQUIRED COMPONENTS marker OPTIONAL_COMPONENTS nvmarker)` provides exactly what I need. Should I assume that the variables in config.mk can just be added into the likwid-config.cmake? E.g. `NVIDIA_INTERFACE` from config.mk --> `set(LIKWID_NVIDIA_INTERFACE @NVIDIA_INTERFACE@)` in likwid-config.cmake

**Comment by TomTheBear** [comment]
That would be a nice feature, wouldn't it. Unfortunately no, it doesn't work like that. Just search for `cmake` in the `Makefile`. There is a `sed` command which replaces `@whatever@` with the actual value ẁhen linking the library:
`@sed -e s+'@PREFIX@'+$(INSTALLED_PREFIX)+g make/likwid-config.cmake > likwid-config.cmake`
The `install` target moves the file to the `lib` directory.

You can use all variables there that are defined in `config.mk`. There are other usable variables in `make/config_*.mk` but they are commonly not that important.

I think I got a much better cmake file a long time ago (wasn't it you?). So maybe we should update the file to be perfect for cmake including all switches that might be required.

**Comment by jrmadsen** [comment]
Thanks #309
